src/components/InputMinusPlus/InputMinusPlus.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div :id="id" class="sol-inputminusplus" :class="{ small }">
+    <button :disabled="newValue === min" @click="decrement">-</button>
+    <input v-model="newValue" type="number" disabled />
+    <button :disabled="newValue === max" @click="increment">+</button>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'InputMinusPlus',
+
+  props: {
+    /** Specify a custom id */
+    id: {
+      type: String,
+      required: true,
+    },
+
+    /** Specify the value of the input - v-model */
+    value: {
+      type: Number,
+      default: 0,
+    },
+
+    /** Specify the minimum value, default 0 */
+    min: {
+      type: Number,
+      default: 0,
+    },
+
+    /** Specify the maximum value */
+    max: {
+      type: Number,
+      required: true,
+    },
+
+    /** Specify whether the Button should be a small */
+    small: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  data() {
+    return {
+      newValue: 0,
+    };
+  },
+
+  watch: {
+    value: function (newVal) {
+      this.newValue = newVal;
+    },
+  },
+
+  created: function () {
+    this.newValue = this.value
+      ? this.value <= this.max
+        ? this.value
+        : this.max
+      : this.min;
```

Comment:
Poderia tirar esses ternários para algo assim

`const actualValue = [this.min, this.value][this.value <= this.max]`
` this.newValue = [this.min, actualValue][this.value]`

Reply:
@alfsouza708 

Acha que faz sentido trocar o ternário por este código?
Para min a leitura ficou pior

`const actualValue = [this.max, this.value][Number(this.value <= this.max)];`
` this.newValue = [this.min, actualValue][Number(!!this.value)];`

Reply:
Realmente, então como alternativa eu jogaria esse ternário na constante pra ter mais legibilidade

`const actualValue = this.value <= this.max ? this.value : this.max;`
`this.newValue = this.value ? actualValue : this.min;`